sourcefiles/ctoptions.py

```diff
@@ -0,0 +1,604 @@
+'''
+Module for preconfiguring in-game options at compile time
+'''
+
+import byteops
+from ctenums import ActionMap, InputMap, LocID
+from ctrom import CTRom
+import ctevent
+
+class ControllerBinds:
+
+    BUTTONS_OFFSET = 0x02FCA9
+    BUTTONS_LENGTH = 8
+
+    def __init__(self, data: bytearray = None):
+
+        self._mappings = self.get_vanilla()
+        
+        if data is None:
+            return
+
+        self.update_from_bytes(data[-8:])
+
+    @classmethod
+    def get_vanilla(cls):
+
+        ret = {
+            ActionMap.CONFIRM: InputMap.A_BUTTON,
+            ActionMap.CANCEL: InputMap.B_BUTTON,
+            ActionMap.MENU: InputMap.X_BUTTON,
+            ActionMap.DASH: InputMap.B_BUTTON,
+            ActionMap.MAP: InputMap.SELECT_BUTTON,
+            ActionMap.WARP: InputMap.Y_BUTTON,
+            ActionMap.PG_UP: InputMap.R_SHOULDER,
+            ActionMap.PG_DN: InputMap.L_SHOULDER
+        }
+        
+        return ret
+
+    def from_rom(self, ctrom: CTRom):
+        rom = ctrom.rom_data
+        
+        rom.seek(BUTTONS_OFFSET)
+        data = rom.read(BUTTONS_LENGTH)
+       
+        return self.to_bytearray(data)
+
+    #return bytearray in order CT expects it in, suitable for writing back to CTRom
+    #optionally, accept an iterable of ints to write in its stead, order not checked from input
+    def to_bytearray(self, data = None):
+    
+        ret = bytearray(8)
+
+        if not data:
```

Comment:
Is there a reason this check isn't `if data is None`?

Reply:
Only that it was the first logically correct check I thought of. Corrected for consistency.